simde/x86/ssse3.h

```diff
@@ -33,6 +33,158 @@ HEDLEY_DIAGNOSTIC_PUSH
 SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
 SIMDE_BEGIN_DECLS_
 
+SIMDE_FUNCTION_ATTRIBUTES
+simde__m128i
+simde_x_mm_deinterleaveeven_epi16 (simde__m128i a, simde__m128i b) {
+  simde__m128i_private
+    r_,
+    a_ = simde__m128i_to_private(a),
+    b_ = simde__m128i_to_private(b);
+  #if defined(SIMDE_ASSUME_VECTORIZATION) && defined(SIMDE_SHUFFLE_VECTOR_)
+    r_.i16 = SIMDE_SHUFFLE_VECTOR_(16, 16, a_.i16, b_.i16, 0, 2, 4, 6, 8, 10, 12, 14);
+  #endif
```

Comment:
This should be an `#else` (and of course you'll need an `#endif` after the loop).  If we have `__builtin_shuffle` or `__builtin_shufflevector` (which is what `SIMDE_SHUFFLE_VECTOR_` maps to) then that one call does everything we need for this function.  In this case, the loop just does it again, but likely slower.

Reply:
oops, fixed this